### ISSUE TITLE
Monitoring - Flux Control Plane Dashboard shows incorrect memory consumption

### ISSUE THREAD

**Issue by pjbgf** [fluxcd/flux2#3086]
During a recent investigation we have noticed that the controller's memory consumption was being incorrectly displayed in our official Dashboard, when compared with the data from [Pod Stats & Info](https://grafana.com/grafana/dashboards/10518-pod-stats-info/) or Pixie:

#### Flux Control Plane > Memory
![image](https://user-images.githubusercontent.com/5452977/189129897-9612ec24-49ee-4e2a-975b-0a6e06fdc532.png)
#### Pod Stats & Info
![image](https://user-images.githubusercontent.com/5452977/189130467-b1301060-37e9-4401-8031-1694e0a7519a.png)
#### Pixie PX/Pod
![image](https://user-images.githubusercontent.com/5452977/189130544-af4d9914-f013-4937-b977-bc566e69559f.png)

We need to review that dashboard and update it accordingly.

**Comment by stefanprodan** [comment]
@pjbgf I would argue that we show the correct stats by looking at Go stats (go_memstats_alloc_bytes) instead of the OS.

**Comment by pjbgf** [comment]
Unfortunately that does not account for CGO or C allocated objects (i.e. libgit2) that were not freed. Changing the current panel (left) to only contain source controller, and adding the same metrics as Pod Stats & Info's I get this for the same time range:
![image](https://user-images.githubusercontent.com/5452977/189139297-ae318d1e-928a-4b86-a39d-a4979d913fe0.png)

If we prefer go metrics, we could use `go_memstats_heap_sys_bytes` instead, which seem more accurate for our use case:
![image](https://user-images.githubusercontent.com/5452977/189142647-73b9f9d9-db69-4c75-a0d8-a5cfb5c2e46b.png)

**Comment by Santosh1176** [comment]
Can I take this up? 
Just to reconfirm, the Metrics in `rate(go_memstats_alloc_bytes_total{namespace=\"$namespace\",pod=~\".*-controller-.*\"}[1m])` should be replaced with the one mentioned by @stefanprodan to resolve this?

**Comment by stefanprodan** [comment]
@Santosh1176 Yes, please note that the `go_memstats_alloc_bytes_total` is a counter while the `go_memstats_heap_sys_bytes` is a gauge.

**Comment by Santosh1176** [comment]
> the go_memstats_heap_sys_bytes is a gauge.


Thankyou @stefanprodan 
Does that mean, rate() shouldn’t be used here?
